### ISSUE TITLE
AJAX request dont work

### ISSUE THREAD

**Issue by mapoxxx** [tcrosen/twitter-bootstrap-typeahead#16]
Hi, 
at first I want to thank you for your great job,
but I have problem and I cannot figure it out.

This is what Ive got:

``` javascript
$('#job-find').typeahead({
        ajax: cakeRoot+'jobFind/getList'
});
```

when I type "Job" plugin send AJAX request successfully, response:

``` javascript
[{"id":"1","name":"Job 1"}{"id":"2","name":"Job 2"}{"id":"3","name":"Job 3"}]
```

but Nothing showed, firebug writes an error:

``` javascript
TypeError: items is null
if (!items.length) {...
in bootstrap-typeahead.js :181
```

content type of response is application/json
I dont know how to fix it.
please help me.

**Comment by tcrosen** [comment]
Hmm it may be a lag issue.  I added an extra check just now to make sure the result isn't null.  Get the latest version and see if it solves your problem.

**Comment by mapoxxx** [comment]
I think problem is in AJAX respond in data variable, becouse firebug is thinking that is string not JSON data.
In line 240:

``` javascript
if (data && data.length && !data[0].hasOwnProperty(that.options.display)) {
return null;
} 
```

Firebug define data[0] as "[" string(char).
Maybe you should define respond type for ajax request as JSON like in jQuery dataType:'json'.
I tried to fix it by myself but I dont know how.

**Comment by mapoxxx** [comment]
Ive changed line 176 
from

``` javascript
this.ajax.data = data;
```

to

``` javascript
this.ajax.data = jQuery.parseJSON(data);
```

and its working, but I think its not very good solution

**Comment by tcrosen** [comment]
jQuery guesses the `dataType` by default, your server method is probably not returning JSON.

If you want to force it without changing the plugin you could always use:

```
$.ajaxSetup({ dataType: 'json' });
```

**Comment by petarvu** [comment]
I had the same issue, which I got fixed by adding display and val properties in addition to the ajax object property. It is not enough to specify the ajax.displayName (seems that it has no effect). Example http://paste2.org/p/2166989

**Comment by tcrosen** [comment]
This will be fixed in 2.0
